src/Microsoft.CodeQuality.Analyzers/CSharp/QualityGuidelines/CSharpRethrowToPreserveStackDetails.cs

```diff
@@ -38,7 +38,9 @@ private static void AnalyzeNode(SyntaxNodeAnalysisContext context)
                                 catchClause.Declaration != null &&
                                 catchClause.Declaration.Identifier.RawKind != 0)
                             {
-                                if (!(context.SemanticModel.GetSymbolInfo(expr).Symbol is ILocalSymbol local) || local.Locations.Length == 0)
+                                if (!(context.SemanticModel.GetSymbolInfo(expr).Symbol is ILocalSymbol local)
+                                    || local.Locations.Length == 0
+                                    || context.SemanticModel.AnalyzeDataFlow(((CatchClauseSyntax)syntax).Block).WrittenInside.Contains(local))
```

Comment:
Kindly also update the VB implementation of the analyzer and add a unit test: https://github.com/dotnet/roslyn-analyzers/blob/d578f2239d4ab331e71b0d5e27f6ffb537176176/src/Microsoft.CodeQuality.Analyzers/VisualBasic/QualityGuidelines/BasicRethrowToPreserveStackDetails.vb#L56-L68

Thanks!

Reply:
I'd like to. But I'm not familiar with vb. And for some reason I can't get the `DataFlowAnalysis` from CatchBlockSyntax. Can someone else help?

Reply:
Tagging @AlekseyTs and @gafter for guidance on `AnalyzeDataFlow` API for VB catch block.
Seems like `AnalyzerDataFlow` method in VB expects a node deriving from `ExecutableStatementSyntax`: http://source.roslyn.io/#Microsoft.CodeAnalysis.VisualBasic/Compilation/SemanticModel.vb,3363, but [CatchBlockSyntax](http://source.roslyn.io/#q=CatchBlockSyntax) derives from `VisualBasicSyntaxNode`

Reply:
Ah, I think you can re-use this overload of the API: http://source.roslyn.io/#Microsoft.CodeAnalysis/Compilation/Extensions.cs,152 pass in the first and last statement of `CatchBlockSyntax.Statements`

Reply:
@mavasani Thanks for the tip! I have updated the commit.